src/lib/layout/hexagons/HexagonGrid.svelte

```diff
@@ -0,0 +1,74 @@
+<script lang="ts">
+  // Inspired by https://github.com/sveltejs/svelte-virtual-list/blob/master/VirtualList.svelte
+  type T = $$Generic;
+
+  export let items: T[];
+  export let cols: number;
+
+  export let gap: 'small' | 'medium' | 'big' = 'medium';
+  export let orientation: 'horizontal' | 'vertical';
+
+  const gridColumnsStyle =
+    orientation === 'horizontal'
+      ? `grid-template-columns: 0.2425fr repeat(${cols}, 0.4805fr 0.2425fr); grid-auto-rows: 0.5fr 0.5fr;`
+      : `grid-template-columns: 0.5fr repeat(${cols}, 0.505fr 0.5fr); grid-auto-rows: 0.251fr 0.5fr;`;
+</script>
+
+<!-- The grid column sizes used are magic numbers that look good with the Hexagon component -->
+<ul class="grid gap-{orientation}-{gap}" style={gridColumnsStyle}>
+  {#each items as item, index}
+    {#if orientation === 'horizontal'}
+      {@const col = index % cols}
+      {@const row = (index - col) / cols}
+      {@const colStart = 2 * col + 1}
+      {@const colEnd = 2 * col + 4}
+      {@const rowStart = 2 * row + 1 + (col % 2)}
+      {@const rowEnd = 2 * row + 3 + (col % 2)}
+      <li style="grid-column: {colStart} / {colEnd}; grid-row: {rowStart} / {rowEnd}">
+        <slot {item} {col} {row} />
+      </li>
+    {:else}
+      {@const col = index % cols}
+      {@const row = (index - col) / cols}
+      {@const rowStart = 2 * row + 1}
+      {@const rowEnd = 2 * row + 4}
+      {@const colStart = 2 * col + 1 + (row % 2)}
+      {@const colEnd = 2 * col + 3 + (row % 2)}
+      <li style="grid-column: {colStart} / {colEnd}; grid-row: {rowStart} / {rowEnd}">
+        <slot {item} {col} {row} />
+      </li>
+    {/if}
```

Comment:
I'm not sure if you will agree with me, but I don't enjoy the duplicated code.
I propose something like this, but I can agree to an extent that the code is a little less readable, but most of this are magic numbers so I think this might be better.
```suggestion
{#each items as item, index}

    {@const isHorizontal = orientation === 'horizontal' }
    {@const col = index % cols}
    {@const row = (index - col) / cols}
    {@const horizontalColStart = 2 * col + 1}
    {@const colStart = horizontalColStart + (isHorizontal ? 0 : (row % 2))}
    {@const verticalRowStart = 2 * row + 1}
    {@const rowStart = verticalRowStart + (isHorizontal ? (col % 2) : 0)}
    {@const colEnd = isHorizontal ? 2 * col + 4 : 2 * col + 3 + (row % 2)}
    {@const rowEnd = isHorizontal ? 2 * row + 3 + (col % 2) : 2 * row + 4}

    <li style="grid-column: {colStart} / {colEnd}; grid-row: {rowStart} / {rowEnd}">
        <slot {item} {col} {row} />
    </li>
```

I think that at least col and row can be extracted from inside elses because they are the same in both conditions, but the rest I leave for your description

Reply:
Thanks for the help! I'll keep looking into other ways to remove code repetition in other files as well.